src/reconcile/rec_write.c

```diff
@@ -2405,20 +2398,35 @@ __rec_split_write(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_REC_CHUNK *chunk
         WT_ASSERT_ALWAYS(session, chunk->entries > 0, "Trying to write an empty chunk");
     }
 
-    if (last_block) {
+    if (last_block && block_meta->page_id != WT_BLOCK_INVALID_PAGE_ID &&
+      block_meta->delta_count < WT_PAGE_DELTA_MAX) {
         WT_RET(__rec_build_delta(session, r, &build_delta));
         /* Discard the delta if it is larger than one tenth of the size of the full image. */
         if (build_delta && r->delta.size > chunk->image.size / 10)
             build_delta = false;
     }
 
     /* Write the disk image and get an address. */
-    WT_RET(__rec_write(session, compressed_image == NULL ? &chunk->image : compressed_image,
-      &chunk->block_meta, addr, &addr_size, &compressed_size, false, F_ISSET(r, WT_REC_CHECKPOINT),
-      compressed_image != NULL));
+    if (build_delta) {
+        /* We must only have one delta. Building deltas for split case is a future thing. */
+        WT_ASSERT(session, last_block);
+        multi->block_meta = *block_meta;
+        ++multi->block_meta.delta_count;
+        ++multi->block_meta.reconciliation_id;
+        WT_RET(__wt_blkcache_write(session, &r->delta, &multi->block_meta, addr, &addr_size,
+          &compressed_size, false, F_ISSET(r, WT_REC_CHECKPOINT), compressed_image != NULL));
+        /* Turn off compression adjustment for delta. */
+        compressed_size = 0;
+    } else {
+        __wt_page_block_meta_init(session, &multi->block_meta);
```

Comment:
The initialization call would erase the metadata, including the base and back-link checkpoint IDs, which are necessary for the page service to process the page update correctly. This is only okay if we are creating a new page (e.g., from a split), but not okay when overwriting an existing page, even if we are writing full image.

Reply:
That being said, we may need to re-initialize page ID on full page write until we get checkpoints working. But after that, we would definitely need to remove this re-initialization.

Reply:
I can do if the page id is invalid, we reinit the block meta.

Reply:
I think the issue is that we should rename the init function to assign function (the next valid page id and the current checkpoint id). We should init the block meta to invalid and that is done when we have a new page that is not reconciled.

Reply:
Sure, feel free to do this instead. I was originally thinking that it would be easier to assign the page ID at the beginning, but if it makes more sense to assign a page ID when the page is first reconciled, then please do that instead. In this way, I presume we can remove the page ID assignments to `WT_BLOCK_INVALID_PAGE_ID`, which would be a big plus. Thanks!